### PR TITLE
feat: create bucket and token from client lib page

### LINKED ISSUE #598: Create Bucket & Token from Client Library Listing Page
**Proposal:**
Allow users to create **Buckets** and **Tokens** from a Client Library listing page.
image
image

**Current behavior:**
When a user is on a Client Library Listing with no bucket they have to leave this view to find where to create a bucket and a token then return to use them.
image
image

**Desired behavior:**
AS a user seeking to use a client library
WHEN I don't the bucket or token that I want to use with the library created yet
THEN I see an `+ Create Bucket` and/or `+ Generate Token` button

WHEN I click the `+ Create Bucket` or `+ Generate Token` button
THEN I see the creation overlay for that resource

WHEN I click the "Tokens" hyperlink
THEN I see the Manage Authentication Tokens documentation page in a new tab (https://docs.influxdata.com/influxdb/cloud/security/tokens/)

WHEN I click the "Buckets hyperlink
THEN I see the Manage Buckets documentation page in a new tab (https://docs.influxdata.com/influxdb/cloud/organizations/buckets/)

**Alternatives considered:**
Add a link to the Buckets and Tokens page.

**Use case:**
This is important for getting users up and running quickly and removing friction from the biggest hurdle to using the application: Loading Data the first time.

### Design Notes
Figma: https://www.figma.com/file/eUoObGYLPu1tZx0omtfIwn/First-Mile?node-id=65%3A0

### PR DESCRIPTION
**Pull request by ShmuelLotman**
Closes #598 

This PR introduces the ability to create a bucket and/or a token in the client libraries page. Previously, the page would display an empty list and the user would be forced to go to the tokens/buckets overlays in their respective tabs and create them, then navigate back to the current page. 

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
Screen Shot 2021-03-08 at 1 17 26 PM
Screen Shot 2021-03-08 at 1 20 01 PM

### CODE REVIEW

File: src/authorizations/components/GenerateTokenDropdown.tsx [hunk 1]
```diff
@@ -8,94 +9,82 @@ import {Dropdown} from '@influxdata/clockface'
...
+      style={{width: '160px'}}
+      button={(active, onClick) => (
+        <Dropdown.Button
+          active={active}
+          onClick={onClick}
+          icon={IconFont.Plus}
+          color={ComponentColor.Primary}
+          testID="dropdown-button--gen-token"
+        >
+          Generate
```
Comment (mavarius) [inline src/authorizations/components/GenerateTokenDropdown.tsx:78]: Let's have this button say "**Generate Token**"

**Review by mavarius — COMMENTED** [review]
Please create & link a follow-up issue to fix the page redirects so we don't lose track of that piece of work.

**Comment by ShmuelLotman** [comment]
@mavarius https://app.zenhub.com/workspaces/applicationui-5e3b05772922e316b3a210a4/issues/influxdata/ui/810

File: src/authorizations/components/GenerateTokenDropdown.tsx [hunk 2]
```diff
@@ -8,94 +9,82 @@ import {Dropdown} from '@influxdata/clockface'
...
-          </Dropdown.Menu>
-        )}
-      />
-    )
+type GenerateTokenProps = RouteComponentProps
+
+const GenerateTokenDropdown: FC<GenerateTokenProps> = ({history}) => {
+  const org = useSelector(getOrg)
+
+  const bucketReadWriteOption = (): string => {
```
Comment (drdelambre) [inline src/authorizations/components/GenerateTokenDropdown.tsx:20]: just make this a string

File: src/authorizations/components/GenerateTokenDropdown.tsx [hunk 3]
```diff
@@ -8,94 +9,82 @@ import {Dropdown} from '@influxdata/clockface'
...
+    return 'Read/Write Token'
   }
 
-  private get optionItems(): JSX.Element[] {
+  const allAccessOption = (): string => {
+    return 'All Access Token'
```
Comment (drdelambre) [inline src/authorizations/components/GenerateTokenDropdown.tsx:25]: make this just a string, not a function that returns a string

File: src/authorizations/components/GenerateTokenDropdown.tsx [hunk 4]
```diff
@@ -8,94 +9,82 @@ import {Dropdown} from '@influxdata/clockface'
...
+          icon={IconFont.Plus}
+          color={ComponentColor.Primary}
+          testID="dropdown-button--gen-token"
+        >
+          Generate Token
+        </Dropdown.Button>
+      )}
+      menu={onCollapse => (
+        <Dropdown.Menu onCollapse={onCollapse}>
+          {getOptionItems()}
```
Comment (drdelambre) [inline src/authorizations/components/GenerateTokenDropdown.tsx:83]: no point really in making this a function either... just put the options in here

File: src/authorizations/components/GenerateTokenDropdown.tsx [hunk 5]
```diff
@@ -8,94 +9,74 @@ import {Dropdown} from '@influxdata/clockface'
...
+          icon={IconFont.Plus}
+          color={ComponentColor.Primary}
+          testID="dropdown-button--gen-token"
+        >
+          Generate Token
+        </Dropdown.Button>
+      )}
+      menu={onCollapse => (
+        <Dropdown.Menu onCollapse={onCollapse}>
+          {[
```
Comment (drdelambre) [inline src/authorizations/components/GenerateTokenDropdown.tsx:56]: remove this line

File: src/authorizations/components/GenerateTokenDropdown.tsx [hunk 6]
```diff
@@ -8,94 +9,74 @@ import {Dropdown} from '@influxdata/clockface'
...
+            <Dropdown.Item
+              testID="dropdown-item generate-token--all-access"
+              id={allAccessOption}
+              key={allAccessOption}
+              value={allAccessOption}
+              onClick={handleSelect}
+            >
+              {allAccessOption}
+            </Dropdown.Item>,
+          ]}
```
Comment (drdelambre) [inline src/authorizations/components/GenerateTokenDropdown.tsx:75]: and this one

**Review by drdelambre — APPROVED** [review]
get rid of those unneeded brackets and the like, then this looks good

File: src/authorizations/components/GenerateTokenDropdown.tsx [hunk 7]
```diff
@@ -8,94 +9,74 @@ import {Dropdown} from '@influxdata/clockface'
...
+          {[
+            <Dropdown.Item
+              testID="dropdown-item generate-token--read-write"
+              id={bucketReadWriteOption}
+              key={bucketReadWriteOption}
+              value={bucketReadWriteOption}
+              onClick={handleSelect}
+            >
+              {bucketReadWriteOption}
+            </Dropdown.Item>,
```
Comment (drdelambre) [inline src/authorizations/components/GenerateTokenDropdown.tsx:65]: and this comma too

**Pull request merged**